### PR TITLE
add quill recipe

### PR DESCRIPTION
**Pull request by odygrd**
Quill is a low latency C++ logging library. The library is header only. The project can be found [here](https://github.com/odygrd/quill/tree/master)

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by odygrd** [comment]
@conda-forge-admin, please update team

**Comment by odygrd** [comment]
@conda-forge/help-c-cpp

File: recipes/quill/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake ${CMAKE_ARGS} \
+  -DCMAKE_PREFIX_PATH:PATH=${PREFIX} \
+  -DCMAKE_INSTALL_PREFIX:PATH=${PREFIX} \
+  -DCMAKE_INSTALL_LIBDIR=lib \
+  -DCMAKE_BUILD_TYPE=Release \
+  .
+
```
Comment (ocefpaf) [inline recipes/quill/build.sh:7]: Is there a `make check` or `ctest` that we can run here and in the bat as well?

File: recipes/quill/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+{% set version = "4.3.0" %}
+{% set sha256 = "c97bf3bfac6dfb7ed77fa08d945a490e302ba07e405539fda61985b39750cb29" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/odygrd/quill/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/quill/meta.yaml:11]: No need to template single use variables. That obscures the recipe. We prefer to keep the jinja at a minimum, just enough to help the auto-updating bot to run.

```suggestion

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://github.com/odygrd/quill/archive/refs/tags/v{{ version }}.tar.gz
  sha256: c97bf3bfac6dfb7ed77fa08d945a490e302ba07e405539fda61985b39750cb29
```

File: recipes/quill/build.sh (hunk 1 above)
Comment (odygrd) [reply recipes/quill/build.sh:10]: I have added it. For mac os there is an error for `MAP_ANONYMOUS` but the correct file `<sys/mman.h>` is already included. Also the repo#s CI passes all tests for macos [here](https://github.com/odygrd/quill/actions/runs/9338908540)

Can you advise on how i can get it resolved so maybe how i can skip running it for macos ?

Comment (odygrd) [reply recipes/quill/build.sh:10]: thanks, is there an `ENV` flag that  I can check inside the build.sh that indicates it is running on macos ?

Comment (ocefpaf) [reply recipes/quill/build.sh:12]: You can probably use an if-clause like:

```shell
if [[ "$(uname)" == "Darwin" ]]; then
  do somthing
fi
``

Comment (odygrd) [reply recipes/quill/build.sh:12]: thanks you, I have done this, I will open an issue on the feedstock repo and will try to resolve it

Comment (odygrd) [reply recipes/quill/build.sh:12]: I've successfully applied a patch to resolve the issue : )

**Pull request merged**